docs/config/index.md

```diff
@@ -131,9 +131,14 @@ export default defineConfig(async ({ command, mode }) => {
 
   For example, `process.env.FOO` and `__APP_VERSION__` are good fits. But `process` or `global` should not be put into this option. Variables can be shimmed or polyfilled instead.
 
+  ::: tip NOTE
+  For using with TypeScript, make sure to add the type declarations in the
+  `env.d.ts` or `vite-env.d.ts` file in order to pass type checking.
```

Comment:
I think both these lines should be in one line following the style of the docs. And some text suggestions:

```suggestion
  For TypeScript users, make sure to add the type declarations in the
  `env.d.ts` or `vite-env.d.ts` file to get type checks and Intellisense.
```

And also maybe a short code block like:

```ts
declare const __APP_VERSION__: string;
```

Reply:
I found myself easier to review and suggest changes for a lone text. But agree that it should match the current style of docs. Updated in https://github.com/vitejs/vite/pull/6153/commits/56c3957ee559ad2ad89406ad9a6b4618fc1d4dff